src/firebase_functions/private/util.py

```diff
@@ -308,3 +309,48 @@ def firebase_config() -> None | FirebaseConfig:
             f'FIREBASE_CONFIG JSON string "{json_str}" is not valid json. {err}'
         ) from err
     return FirebaseConfig(storage_bucket=json_data.get("storageBucket"))
+
+
+def nanoseconds_timestamp_conversion(time: str) -> _dt.datetime:
+    """Converts a nanosecond timestamp and returns a datetime object of the current time in UTC"""
+
+    # Split the string into date-time and nanoseconds
+    s_datetime, s_ns = time.split(".")
+
+    # Split the nanoseconds from the timezone specifier ('Z')
+    s_ns, _ = s_ns.split("Z")
+
+    # Only take the first 6 digits of the nanoseconds
+    s_ns = s_ns[:6]
+
+    # Put the string back together
+    s_processed = f"{s_datetime}.{s_ns}Z"
+
+    # Now parse the date-time string
+    event_time = _dt.datetime.strptime(s_processed, "%Y-%m-%dT%H:%M:%S.%fZ")
+
+    # strptime assumes local time, we know it's UTC and so:
+    event_time = event_time.replace(tzinfo=_dt.timezone.utc)
```

Comment:
```suggestion
  # Separate the date and time part from the nanoseconds.
  datetime_str, nanosecond_str = timestamp_string.replace("Z", "").split(".")

  # Parse the date and time part of the string.
  event_time = datetime.strptime(datetime_str, "%Y-%m-%dT%H:%M:%S")

  # Add the microseconds and timezone.
  event_time = event_time.replace(microsecond=int(nanosecond_str[:6]), tzinfo=timezone.utc)
```

we could probably simplify to something like this